source/databricks/calculation_engine/tests/calculation/energy/aggregations/exchange_aggregators/test_aggregate_net_exchange_per_ga.py

```diff
@@ -60,21 +60,109 @@ def test_returns_distinct_qualities_from_both_to_and_from_ga(
         expected_qualities: list[QuantityQuality],
     ) -> None:
         # Arrange
+        from_grid_area = "111"
+        to_grid_area = "222"
         rows = [
-            *[factories.create_to_row(quality=quality) for quality in to_ga_qualities],
             *[
-                factories.create_from_row(quality=quality)
+                factories.create_to_row(
+                    quality=quality,
+                    grid_area=to_grid_area,
+                    from_grid_area=from_grid_area,
+                )
+                for quality in to_ga_qualities
+            ],
+            *[
+                factories.create_from_row(
+                    quality=quality,
+                    grid_area=from_grid_area,
+                    to_grid_area=to_grid_area,
+                )
                 for quality in from_ga_qualities
             ],
         ]
         metering_point_time_series = factories.create(spark, rows)
         expected_qualities = sorted([q.value for q in expected_qualities])
 
         # Act
-        actual = aggregate_net_exchange_per_ga(metering_point_time_series)
+        actual = aggregate_net_exchange_per_ga(
+            metering_point_time_series, [from_grid_area, to_grid_area]
+        )
 
         # Assert
         actual_rows = actual.df.collect()
         assert len(actual_rows) == 2
         assert sorted(actual_rows[0][Colname.qualities]) == expected_qualities
         assert sorted(actual_rows[1][Colname.qualities]) == expected_qualities
+
+
+class TestWhenMeteringPointIsNeitherInToOrFromGridArea:
+    def test_returns_result_with_only_to_and_from_grid_area(
+        self,
+        spark: SparkSession,
+    ) -> None:
+        # Arrange
+        grid_area_1 = "123"  # this is the grid area of the metering point
+        grid_area_2 = "234"
+        grid_area_3 = "345"
```

Comment:
Perhaps we could make the names more expressive? E.g.
```suggestion
        other_grid_area = "123"  # this is the grid area of the metering point
        exchange_grid_area_1 = "234"
        exchange_grid_area_2 = "345"
```